packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -84,6 +84,49 @@ const Duration _kCursorBlinkHalfPeriod = Duration(milliseconds: 500);
 // is shown in an obscured text field.
 const int _kObscureShowLatestCharCursorTicks = 3;
 
+class _RenderCompositionCallback extends RenderProxyBox {
+  _RenderCompositionCallback(this.compositeCallback);
+
+  final void Function(Layer) compositeCallback;
+
+  bool get enabled => _enabled;
+  bool _enabled = false;
+  VoidCallback? _cancelCallback;
+  set enabled(bool newValue) {
```

Comment:
For my own understanding: If this is set to true, the callback will not actually get called until this object is repainted. So the callback may miss some composition phases, if these happen without repainting the object. Is that the expected behavior (seems confusing)? Should we force a repaint if you enable it?